kornia/geometry/camera/stereo.py

```diff
@@ -249,38 +269,28 @@ def reproject_disparity_to_3D(disparity_tensor: torch.Tensor, Q_matrix: torch.Te
     _check_Q_matrix(Q_matrix)
     _check_disparity_tensor(disparity_tensor)
 
-    batch_size, rows, cols = disparity_tensor.shape
+    batch_size, channels, rows, cols = disparity_tensor.shape
     dtype = disparity_tensor.dtype
     device = disparity_tensor.device
-    homogenous_observation_ndim = 4
-    euclidian_observation_ndim = homogenous_observation_ndim - 1
-
-    # Construct a mesh grid of uv values, i.e. the tensors will contain 1:rows and 1:cols.
-    u, v = torch.meshgrid(
-        torch.arange(rows, dtype=dtype, device=device), torch.arange(cols, dtype=dtype, device=device)
-    )
-    u, v = u.expand(batch_size, -1, -1), v.expand(batch_size, -1, -1)
 
-    # The z dimension in homogenous coordinates are just 1.
-    z = torch.ones((batch_size, rows, cols), dtype=dtype, device=device)
+    uv = create_meshgrid(rows, cols, normalized_coordinates=False, device=device, dtype=dtype)
+    v, u = uv[..., 0], uv[..., 1]
+    u, v = u.expand(batch_size, channels, -1, -1), v.expand(batch_size, channels, -1, -1)
 
     # Stack the observations into a tensor of shape (batch_size, 4, -1) that contains all
     # 4 dimensional vectors [u v disparity 1].
-    uvdz = torch.stack((u, v, disparity_tensor, z), 1).reshape(batch_size, homogenous_observation_ndim, -1)
+    uvd = torch.stack((u, v, disparity_tensor), 1).reshape(batch_size, 3, -1).permute(0, 2, 1)
```

Comment:
another trick you could do here to avoid permute and homogeneous conversion is using `transform_points`
https://kornia.readthedocs.io/en/latest/geometry.linalg.html?highlight=transform_points#kornia.geometry.linalg.transform_points

```python
uv = uv.view(batch_size, -1, -1, -1)  # BxHxWx2
uvd = torch.stack((uv[..., 0], uv[..., 1], disparity_tensor[:, 0]), dim=-1)  # BxHxWx3
points = transform_points(Q_matrix, uvd)  # BxHxWx3
```
you might have to play with `Q_matrix` unsqueeze

Reply:
Handy function!